src/providers/mssql/qgsmssqltablemodel.cpp

```diff
@@ -370,6 +377,16 @@ QString QgsMssqlTableModel::layerURI( const QModelIndex &index, const QString &c
 
   uri.setParam( QStringLiteral( "disableInvalidGeometryHandling" ), disableInvalidGeometryHandling ? QStringLiteral( "1" ) : QStringLiteral( "0" ) );
 
+  if ( QgsMssqlConnection::geometryColumnsOnly( mConnectionName ) )
+  {
+    uri.setParam( QStringLiteral( "extentInGeometryColumns" ), QgsMssqlConnection::extentInGeometryColumns( mConnectionName ) ? QStringLiteral( "1" ) : QStringLiteral( "0" ) );
+  }
+
+  QStandardItem *isViewItem = itemFromIndex( index.sibling( index.row(), DbtmView ) );
+
+  if ( isViewItem->data( Qt::UserRole + 1 ).toBool() )
+    uri.setParam( QStringLiteral( "primaryKeyInGeometryColumns" ), QgsMssqlConnection::primaryKeyInGeometryColumns( mConnectionName ) ? QStringLiteral( "1" ) : QStringLiteral( "0" ) );
```

Comment:
Shouldn't we allow using primaryKeyInGeometryColumns only when QgsMssqlConnection::geometryColumnsOnly( mConnectionName ) == true ?

Reply:
see other comment